### PR TITLE
[DOC] Tweaks for Array#sort

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: array.c [hunk 1]
```diff
@@ -3455,44 +3455,34 @@ rb_ary_sort_bang(VALUE ary)
...
- *  (see Comparable):
+ *  With no block given, compares elements using operator <tt>#<=></tt>
+ *  (see Object#<=>):
  *
- *    a = 'abcde'.split('').shuffle
- *    a # => ["e", "b", "d", "a", "c"]
- *    a1 = a.sort
- *    a1 # => ["a", "b", "c", "d", "e"]
+ *    a = [0, 1, 2, 3].shuffle # => [0, 2, 3, 1]
+ *    a.sort                   # => [0, 1, 2, 3]
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply array.c]: Fixed.

File: array.c [hunk 2]
```diff
@@ -3455,44 +3455,34 @@ rb_ary_sort_bang(VALUE ary)
...
  *
- *    a = 'abcde'.split('').shuffle
- *    a # => ["e", "b", "d", "a", "c"]
- *    a1 = a.sort {|a, b| a <=> b }
- *    a1 # => ["a", "b", "c", "d", "e"]
- *    a2 = a.sort {|a, b| b <=> a }
- *    a2 # => ["e", "d", "c", "b", "a"]
+ *    a = [0, 1, 2, 3].shuffle # => [3, 2, 0, 1]
+ *    a.sort {|a, b| a <=> b } # => [0, 1, 2, 3]
+ *    a.sort {|a, b| b <=> a } # => [3, 2, 1, 0]
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply array.c]: Fixed.

File: array.c [hunk 3]
```diff
@@ -3455,44 +3455,34 @@ rb_ary_sort_bang(VALUE ary)
...
- *    a = 'abcde'.split('').shuffle
- *    a # => ["e", "b", "d", "a", "c"]
- *    a1 = a.sort
- *    a1 # => ["a", "b", "c", "d", "e"]
+ *    a = [0, 1, 2, 3].shuffle # => [0, 2, 3, 1]
+ *    a.sort                   # => [0, 1, 2, 3]
  *
- *  With a block, calls the block with each element pair;
- *  for each element pair +a+ and +b+, the block should return an integer:
+ *  With a block given, calls the block with each #combination of pairs of elements from +self+;
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply array.c]: Not a typo;  clarified(?) to make clear that the link leads to a discussion of _combination_.

File: array.c [hunk 4]
```diff
@@ -3455,44 +3455,34 @@ rb_ary_sort_bang(VALUE ary)
...
- *    a = 'abcde'.split('').shuffle
- *    a # => ["e", "b", "d", "a", "c"]
- *    a1 = a.sort
- *    a1 # => ["a", "b", "c", "d", "e"]
+ *    [0, 2, 3, 1].sort # => [0, 1, 2, 3]
  *
- *  With a block, calls the block with each element pair;
- *  for each element pair +a+ and +b+, the block should return an integer:
+ *  With a block given, calls the block with each combination of pairs of elements from +self+
+ *  (see Array#combination);
```
Comment (peterzhu2118) [inline array.c:3469]: I don't think it is necessary refer to the `Array#combination` method. This has nothing to do with that method but rather only have to do with the mathematical concept of combinations.

Comment (BurdetteLamar) [reply array.c]: Removed.

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**